ArduCopter/mode.h

```diff
@@ -1392,3 +1392,93 @@ class ModeZigZag : public Mode {
 
     uint32_t reach_wp_time_ms = 0;  // time since vehicle reached destination (or zero if not yet reached)
 };
+
+#if MODE_AUTOROTATE_ENABLED == ENABLED
+class ModeAutorotate : public Mode {
+
+public:
+
+    // inherit constructor
+    using Mode::Mode;
+
+    bool init(bool ignore_checks) override;
+    void run() override;
+
+    bool is_autopilot() const override { return true; }
+    bool requires_GPS() const override { return false; }
+    bool has_manual_throttle() const override { return false; }
+    bool allows_arming(bool from_gcs) const override { return false; };
+
+    static const struct AP_Param::GroupInfo  var_info[];
+
+protected:
+
+    const char *name() const override { return "AUTOROTATE"; }
+    const char *name4() const override { return "AROT"; }
+
+private:
+
+    // --- Internal variables ---
+    float _initial_rpm;             //Head speed recorded at initiation of flight mode (RPM)
+    float _target_head_speed;       //The terget head main rotor head speed.  Normalised by main rotor set point
+    float _fwd_speed_target;        //Target forward speed (cm/s)
+    float _desired_v_z;             //Desired vertical
+    int32_t _pitch_target;          //Target pitch attitude to pass to attitude controller
+    float _collective_aggression;   //The 'aggresiveness' of collective appliction
+    float _z_touch_down_start;      //The height in cm that the touch down phase began
+    float _t_touch_down_initiate;   //The time in ms that the touch down phase began
+    float now;                      //Time at start of cycle
```

Comment:
now is a very commonly used local variable to mean the current time.  if it's something else maybe change the name?

Reply:
This variable is the same as the common use.  I have reworded the comment to make that clear.